recipe/activate.bat

```diff
@@ -0,0 +1,26 @@
+:: Check whether there are dlls for openssl on the system path that would gets
+:: picked up by the windows loader before those in the conda environment.
+:: If yes, warn that the environment is potentially vulnerable.
+
+@echo off
+
+set "HAS_SYS_LIBS=F"
+set "HAS_SYS_SSL=F"
+set "HAS_SYS_CRYPTO=F"
+if exist C:\Windows\System32\libssl-1_1-x64.dll (
+  set "HAS_SYS_LIBS=T"
+  set "HAS_SYS_SSL=T"
+)
+if exist C:\Windows\System32\libcrypto-1_1-x64.dll (
+  set "HAS_SYS_LIBS=T"
+  set "HAS_SYS_CRYPTO=T"
+)
+
+:: The carets are used for escaping brackets, which would otherwise be interpreted (and fail).
+if "%HAS_SYS_LIBS%"=="T" (
+                             ECHO WARNING: Your system contains ^(potentially^) outdated libraries under:
+  if "%HAS_SYS_SSL%"=="T"    ECHO WARNING: %LIBSSL_PATH%
+  if "%HAS_SYS_CRYPTO%"=="T" ECHO WARNING: %LIBCRYPTO_PATH%
+                             ECHO WARNING: These libraries will be linked before those in the conda
+                             ECHO WARNING: environment and might make your installation vulnerable!
```

Comment:
Add a message about `CONDA_DLL_SEARCH_MODIFICATION_ENABLE` as well?

Reply:
Well yes, but even if they do set it, we leave the user guessing whether it worked or not, which is far from ideal.

I would have liked to do that (set the envvar and try again) within the activation script, but I cannot figure out a way to reliably determine which dll get's picked up, short of loading `cryptography` (obviously not available in this recipe) and seeing which backend gets loaded.

Reply:
How about doing that `CONDA_DLL_SEARCH_MODIFICATION_ENABLE` check upon activation in the `cryptography` feedstock?

For openssl it wouldn't help necessarily anyway, that is - if a package wants to load the dll *from outside of python*, then it would still pick up the one in `C:\Windows\System32`. Only within a correctly adapted python (cf. https://github.com/conda-forge/python-feedstock/pull/345) would the `CONDA_DLL_SEARCH_MODIFICATION_ENABLE` adapt the windows loader correctly, IIUC.

Reply:
Ping @isuruf 
What do you think of my last comment?

Reply:
Makes sense

Reply:
I love these changes!